iroha/src/lib.rs

```diff
@@ -293,9 +307,58 @@ impl Iroha {
             sumeragi_message_handle,
             tx_handle,
             block_sync_message_handle,
+            genesis_block_handle,
         );
         Ok(())
     }
+
+    /// To wait for active peers(this is provisional code)
+    /// Wait 500ms, 1000ms, 1500ms, ...
+    async fn wait_for_active_peers(
+        sorted_peers: &[PeerId],
+        myself_peer_id: &PeerId,
+    ) -> Result<(), String> {
+        log::info!("Waiting for active peers");
+        let retry_count = 10;
+        let retry_duration_ms = 500;
+        for peer in sorted_peers.iter() {
+            let mut failed_to_connect = true;
+            for i in 0..=retry_count {
+                if peer == myself_peer_id {
+                    failed_to_connect = false;
+                    break;
+                }
+                let target = http_uri::Builder::new()
+                    .scheme("http")
+                    .authority(peer.address.as_str())
+                    .build()
+                    .unwrap_or_else(|_| panic!("Invalid peer address {}", peer.address.as_str()));
+                match Network::send_request_to(
+                    &target.to_string(),
+                    Request::new(uri::HEALTH_URI.to_string(), Vec::new()),
+                )
```

Comment:
```suggestion
                match Network::send_request_to(
                    peer.address,
                    Request::new(uri::HEALTH_URI.to_string(), Vec::new()),
                )
```

It is no longer an http request.